### ISSUE TITLE
Building as Release/RelWithDebInfo will throw a warning (warnings are treated as errors)

### ISSUE THREAD

**Issue by johnzupin** [KhronosGroup/SPIRV-Reflect#92]
Building using cmake version 3.16.3 and setting the build type flag as `-DCMAKE_BUILD_TYPE=Release` will throw a warning.

```
[ 29%] Building CXX object CMakeFiles/spirv-reflect.dir/common/output_stream.cpp.o
/home/john/Development/SPIRV-Reflect/common/output_stream.cpp: In function ‘void WriteReflection(const spv_reflect::ShaderModule&, bool, std::ostream&)’:
/home/john/Development/SPIRV-Reflect/common/output_stream.cpp:945:20: error: variable ‘result’ set but not used [-Werror=unused-but-set-variable]
  945 |  
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/spirv-reflect.dir/build.make:115: CMakeFiles/spirv-reflect.dir/common/output_stream.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:121: CMakeFiles/spirv-reflect.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by johnzupin** [comment]
PR https://github.com/KhronosGroup/SPIRV-Reflect/pull/91 fixes this but another warning shows up after it:

```
[ 83%] Building C object examples/CMakeFiles/io_variables.dir/__/spirv_reflect.c.o
/home/john/Development/SPIRV-Reflect/common/output_stream.cpp: In function ‘void WriteReflection(const spv_reflect::ShaderModule&, bool, std::ostream&)’:
/home/john/Development/SPIRV-Reflect/common/output_stream.cpp:951:20: error: variable ‘result’ set but not used [-Werror=unused-but-set-variable]
  951 |   SpvReflectResult result = obj.EnumerateInputVariables(&count, nullptr);
      |                    ^~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/spirv-reflect.dir/build.make:115: CMakeFiles/spirv-reflect.dir/common/output_stream.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:121: CMakeFiles/spirv-reflect.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by johnzupin** [comment]
I should probably also mention that I am using Ubuntu 18.04

**Comment by chaoticbob** [comment]
Hi John,

Which compiler and what version of it are you using? I just tried this on clang 10 and gcc 9.3 and couldn't reproduce.

Thanks

**Comment by johnzupin** [comment]
I'm using `gcc (Ubuntu 9.3.0-10ubuntu2) 9.3.0` on my Ubuntu desktop.

I also testing compiling the repo as release on my laptop that has Slackware installed, which uses gcc 9.3.0 also but cmake version 3.17.0, and while the same warning was present it did not treat it as an error...

**Comment by johnzupin** [comment]
I'm using gcc (Ubuntu 9.3.0-10ubuntu2) 9.3.0 on my Ubuntu desktop.

I also tested compiling the repo as release on my laptop that has Slackware installed, which uses gcc 9.3.0 also but cmake version 3.17.0, and while the warning was present it did not treat it as an error...

**Comment by johnzupin** [comment]
There is no warning if you build Debug. Are you passing the build type to cmake as `-DCMAKE_BUILD_TYPE=Release`?

**Comment by greg-lunarg** [comment]
I have just pushed some additional code onto #91. @johnzupin does this fix your problem?

**Comment by johnzupin** [comment]
@greg-lunarg  Yes that seems to fix my problem.

**Comment by johnzupin** [comment]
> I also tested compiling the repo as release on my laptop that has Slackware installed, which uses gcc 9.3.0 also but cmake version 3.17.0, and while the warning was present it did not treat it as an error...

Sorry for any confusion, but I am seeing the warning being treated as an error on cmake 3.17.0. My build environment wasn't clean.

**Comment by greg-lunarg** [comment]
Are you sure you got the latest code? If so, can you please post the warning/error message?

**Comment by johnzupin** [comment]
> Are you sure you got the latest code? If so, can you please post the warning/error message?

Yeah your PR works, my other comment was to correct a mistake I made before about a cmake version not treating warnings as errors.

**Comment by johnzupin** [comment]
Closing, issue fixed with PR's #91  #93
